src/NBitcoin/BlockHeader.cs

```diff
@@ -98,7 +98,12 @@ public virtual void ReadWrite(BitcoinStream stream)
         /// <summary>Populates stream with items that will be used during hash calculation.</summary>
         protected virtual void ReadWriteHashingStream(BitcoinStream stream)
         {
-            this.ReadWrite(stream);
+            stream.ReadWrite(ref this.version);
```

Comment:
if this.readwrite doesn't work for your maybe we should move those lines to separated method and call this method from readwrite and readwriteHashingStream